fili-core/src/test/groovy/com/yahoo/bard/webservice/web/handlers/CacheWeightCheckRequestHandlerSpec.groovy

```diff
@@ -0,0 +1,545 @@
+
+import com.fasterxml.jackson.core.JsonFactory
+import com.fasterxml.jackson.core.JsonParser
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.node.JsonNodeFactory
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.ObjectWriter
+
+import javax.ws.rs.container.ContainerRequestContext
+import javax.ws.rs.core.MultivaluedHashMap
+
+import spock.lang.Specification
+
+class CacheWeightCheckRequestHandlerSpec extends Specification {
+    private static final ObjectMapper MAPPER = new ObjectMappersSuite().getMapper()
+
+    DataRequestHandler next
+    DruidWebService webService
+    QueryWeightUtil queryWeightUtil
+    ObjectWriter writer
+    TupleDataCache<String, Long, String> dataCache
+    QuerySigningService<Long> querySigningService
+    CacheService cacheService
+    CacheWeightCheckRequestHandler handler
+
+    RequestContext context
+    DataApiRequest request
+    GroupByQuery groupByQuery
+    ResponseProcessor response
+
+    BardQueryInfo bardQueryInfo
+    JsonNode json
+
+    def setup() {
+        next = Mock(DataRequestHandler)
+        webService = Mock(DruidWebService)
+        queryWeightUtil = Mock(QueryWeightUtil)
+        writer = Mock(ObjectWriter)
+        MAPPER.writer() >> writer
+        context = Mock(RequestContext)
+        request = Mock(DataApiRequest)
+        dataCache = Mock(TupleDataCache)
+        querySigningService = Mock(SegmentIntervalsHashIdGenerator)
+        groupByQuery = Mock(GroupByQuery)
+        groupByQuery.getInnermostQuery() >> groupByQuery
+        response = Mock(WeightCheckResponseProcessor)
+        bardQueryInfo = BardQueryInfoUtils.initializeBardQueryInfo()
+        json = new JsonNodeFactory().arrayNode()
+        cacheService = new CacheService(dataCache, querySigningService, MAPPER)
+        handler = new CacheWeightCheckRequestHandler(
+                next,
+                webService,
+                queryWeightUtil,
+                MAPPER,
+                cacheService
+        )
+    }
+
+    def cleanup() {
+        BardQueryInfoUtils.resetBardQueryInfo()
+    }
+
+    def "Test constructor"() {
+        expect:
+        handler.next == next
+        handler.webService == webService
+        handler.queryWeightUtil == queryWeightUtil
+        handler.mapper == MAPPER
+        handler.cacheService == cacheService
+    }
+
+    def "Test handle request with request passing quick weight check"() {
+        setup:
+        1 * queryWeightUtil.skipWeightCheckQuery(groupByQuery) >> true
+        1 * next.handleRequest(context, request, groupByQuery, response) >> true
+
+        expect:
+        handler.handleRequest(context, request, groupByQuery, response)
+
+        and:
+        bardQueryInfo.queryCounter.get(BardQueryInfo.WEIGHT_CHECK).get() == 0
+    }
+
+    def "Test handleRequest without building callback"() {
```

Comment:
The name should probably reflect which code path this represents.  Is this case is this 'read cache is true but cache is a miss'?